### DISCUSSION TITLE
Naming convention (remember)

### DISCUSSION THREAD

**Discussion by InsaneDoggo** [badoo/Decompose#127] [General]
Function `asState` should be renamed to `rememberAsState`

https://github.com/arkivanov/Decompose/blob/master/extensions-compose-jetbrains/src/commonMain/kotlin/com/arkivanov/decompose/extensions/compose/jetbrains/ValueComposable.kt#L12

Acording to Compose's naming conventions:
`prefix any @Composable factory function that internally remember {}s and returns a mutable object with the prefix remember.`

[Deatils](https://github.com/androidx/androidx/blob/androidx-main/compose/docs/compose-api-guidelines.md#naming-composable-functions-that-remember--the-objects-they-return)

**Comment by arkivanov** [comment]
Thanks for the idea. But it doesn't return a mutable object. Perhaps something like `listenAsState` or `subscribeAsState`, similar to `Flow.collectAsState`.

**Comment by arkivanov** [comment]
Filed #128

**Comment by Krishprajapati15** [comment]
Hey 👋

According to [Jetpack Compose’s naming conventions](https://developer.android.com/jetpack/compose/mental-model#state), any @Composable factory function that uses remember {} internally and returns a mutable object should be prefixed with remember.
So I’d suggest renaming asState to rememberAsState for better alignment and clarity:

✅ Follows Compose conventions

✅ Makes the function’s behavior more obvious to developers using it

Here’s the link to the function in question for quick reference:
https://github.com/arkivanov/Decompose/blob/master/extensions-compose-jetbrains/src/commonMain/kotlin/com/arkivanov/decompose/extensions/compose/jetbrains/ValueComposable.kt#L12

Happy to help with a PR if needed!

**Reply by arkivanov** [reply]
The function was renamed to `subscribeAsState` a long time ago. Your link is broken as the `ValueComposable.kt` file and the entire `extensions-compose-jetbrains` folder no longer exist. See: [SubscribeAsState.kt](https://github.com/arkivanov/Decompose/blob/master/extensions-compose/src/commonMain/kotlin/com/arkivanov/decompose/extensions/compose/SubscribeAsState.kt).

Please note that I'm not maintaining this repository, see https://github.com/arkivanov/Decompose.

**Comment by Krishprajapati15** [comment]
Hey 👋

Thanks for the clarification!

You're absolutely right — I hadn't noticed that the function has since been renamed to `subscribeAsState`, and the original file structure is no longer valid. Appreciate the updated link and context.

I'll make sure to check the latest state of the repo next time before referencing a path 😅

Thanks again for pointing that out and for maintaining such a valuable tool for the Compose ecosystem!

Cheers ✌️
